src/ShopBundle/objects/db/TShopOrderStep/TShopOrderStep.class.php

```diff
@@ -150,7 +150,7 @@ public static function GetNewInstance($sData = null, $sLanguage = null)
             $class = $sData['class'];
         }
         $shopBasket = TShopBasket::GetInstance();
-        $object = null;
+        $object = new TdbShopOrderStep(null, $sLanguage);
```

Comment:
`GetNewInstance` should always create a class. The Class may not contain any data because no record was loaded, but it should always be there. 

The signature used to indicate this as well - even if the source could return null if there was no session. Instead of changing the signatur to allow null, the method code should have been changed. 

If searched for usage - they all assume that the method can never return null - so I think we are safer avoiding null.